packages/perseus-editor/src/components/angle-input.tsx

```diff
@@ -0,0 +1,119 @@
+import * as KAS from "@khanacademy/kas";
+import {components} from "@khanacademy/perseus";
+import {View} from "@khanacademy/wonder-blocks-core";
+import {TextField} from "@khanacademy/wonder-blocks-form";
+import {Strut} from "@khanacademy/wonder-blocks-layout";
+import Switch from "@khanacademy/wonder-blocks-switch";
+import {spacing} from "@khanacademy/wonder-blocks-tokens";
+import {LabelMedium, LabelSmall} from "@khanacademy/wonder-blocks-typography";
+import {StyleSheet} from "aphrodite";
+import * as React from "react";
+
+const {InfoTip} = components;
+
+const degreeToRadian = (degrees: number) => {
+    return (degrees / 180) * Math.PI;
+};
+
+type Props = {
+    angle: number;
+    onChange: (angle: number) => void;
+};
+
+const AngleInput = (props: Props) => {
+    const {angle, onChange} = props;
+
+    const [angleInput, setAngleInput] = React.useState(angle.toString());
+    const [angleType, setAngleType] = React.useState<"degrees" | "radians">(
+        "radians",
+    );
+
+    function handleAngleChange(newValue) {
+        // Update the local state (update the input field value).
+        setAngleInput(newValue);
+
+        try {
+            // If the new value is a valid expression, update the props.
+            // Save the angle in radians.
+            const evaluatedAngle = KAS.parse(newValue).expr.eval();
+
+            if (angleType === "degrees") {
+                onChange(degreeToRadian(evaluatedAngle));
+            } else {
+                onChange(evaluatedAngle);
+            }
+        } catch (e) {
+            // The user likely has not finished typing the expression.
+            // Do nothing.
+            return;
+        }
+    }
+
+    function handleAngleTypeChange() {
+        // Update the angle to the new type.
+        const evaluatedAngle = KAS.parse(angleInput).expr.eval();
```

Comment:
Ah yes, you're right! I'm thinking of streamlining this by calling `handleAngleChange` within `handleAngleTypeChange()`

Reply:
Okay... I did it. But now I'm worried that if someone does type in "2p" instead of "2pi" and then press the switch, they could be confused by there being no changes at all.

Maybe I'll revisit validation later. I think I have to add a bunch of validation stuff for all the other locked figures too anyway.